### PR TITLE
feat: 🎸 CRUDL capabilites for ldap managed-group

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: ICU-10100
✅ Closes: ICU-10099
✅ Closes: ICU-10101
✅ Closes: ICU-10102

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10100)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10099)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10101)
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10102)

## Description
Adds crudl capabilities for managed groups. Apologies for the chunky PR, I will do better next time at separating it into different PRs.

:technologist: [Admin preview](https://boundary-ui-git-icu-10100-ldap-managed-group-create-hashicorp.vercel.app/scopes/global/auth-methods/am_ec8st2tp51/managed-groups)

### CODE REVIEW

File: ui/admin/app/components/auth-methods/auth-method/managed-groups/managed-group/nav/index.hbs [hunk 1]
```diff
@@ -9,9 +9,11 @@
   >
     {{t 'titles.details'}}
   </nav.link>
-  <nav.link
-    @route='scopes.scope.auth-methods.auth-method.managed-groups.managed-group.members'
-  >
-    {{t 'resources.managed-group.member.title_plural'}}
-  </nav.link>
```
Comment (lisbet-alvarez) [inline ui/admin/app/components/auth-methods/auth-method/managed-groups/managed-group/nav/index.hbs:16]: Temporarily, hiding the members tab

File: addons/api/addon/serializers/managed-group.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ import {
 
 export default class ManagedGroupSerializer extends ApplicationSerializer {
   // =properties
+  /**
+   * @type {boolean}
+   */
+  serializeScopeID = false;
```
Comment (lisbet-alvarez) [inline addons/api/addon/serializers/managed-group.js:17]: the scope_id should not be sent as part of the request body for managed-groups

Comment (DhariniJeeva) [reply addons/api/addon/serializers/managed-group.js:17]: ty for adding this back!

File: ui/admin/app/components/form/managed-group/index.hbs [hunk 3]
```diff
@@ -3,81 +3,14 @@
...
+  {{component
+    (concat 'form/managed-group/' @model.type)
+    model=@model
+    edit=@edit
+    submit=@submit
+    cancel=@cancel
+    removeItemByIndex=@removeItemByIndex
+    addStringItem=@addStringItem
+  }}
+{{/if}}
```
Comment (dvonanderson) [inline ui/admin/app/components/form/managed-group/index.hbs:16]: @lisbet-alvarez very nice refactor

**Review by dvonanderson — APPROVED** [review]
LGTM

**Comment by DhariniJeeva** [comment]
Seeing [group_names](https://boundary-jtjomgg23-hashicorp.vercel.app/scopes/global/auth-methods/am_e4nm52bkf2/managed-groups) for OIDC managed group, the field should be only visible for LDAP

**Comment by lisbet-alvarez** [comment]
> Seeing [group_names](https://boundary-jtjomgg23-hashicorp.vercel.app/scopes/global/auth-methods/am_e4nm52bkf2/managed-groups) for OIDC managed group, the field should be only visible for LDAP

@DhariniJeeva great catch! fixed! thank you!

**Review by DhariniJeeva — APPROVED** [review]

**Review by cameronperera — APPROVED** [review]

**Pull request merged**